src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -27372,5 +27368,554 @@ public ValueTuple(T1 item1)
+            Assert.Equal(0, item1.TupleElementIndex);
+
+            var item1Underlying = item1.TupleUnderlyingField;
+            Assert.IsType<SubstitutedFieldSymbol>(item1Underlying);
+            Assert.Equal("System.Int32 (System.Int32 Item1, System.Int32 Item2).Item1", item1Underlying.ToTestDisplayString());
+            Assert.Equal(0, item1Underlying.TupleElementIndex);
+            Assert.Same(item1Underlying, item1Underlying.TupleUnderlyingField);
+
+            var item2 = tuple.GetMember<TupleElementFieldSymbol>("Item2");
+            Assert.Equal(1, item2.TupleElementIndex);
+            var item2Underlying = item2.TupleUnderlyingField;
+            Assert.IsType<SubstitutedFieldSymbol>(item2Underlying);
+            Assert.Equal(1, item2Underlying.TupleElementIndex);
+            Assert.Same(item2Underlying, item2Underlying.TupleUnderlyingField);
+        }
+
+        [Fact]
+        public void IndexOfUnderlyingFieldsInTuple9()
+        {
+            string source = @"
+class C
+{
+    (int, int, int, int, int, int, int, int, int) M() => throw null;
+}
+";
+            var comp = CreateCompilation(source);
+            var m = (MethodSymbol)comp.GetMember("C.M");
+            var tuple = m.ReturnType;
+            Assert.Equal("(System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32)",
+                tuple.ToTestDisplayString());
+            Assert.IsType<ConstructedNamedTypeSymbol>(tuple);
+
+            var item8 = tuple.GetMember<TupleElementFieldSymbol>("Item8");
+            Assert.Equal("System.Int32 (System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32).Item8",
+                item8.ToTestDisplayString());
+            Assert.Equal(7, item8.TupleElementIndex);
+
+            var item8Underlying = item8.TupleUnderlyingField;
+            Assert.Equal("Item1", item8Underlying.Name);
+            Assert.IsType<SubstitutedFieldSymbol>(item8Underlying);
+            Assert.Equal(0, item8Underlying.TupleElementIndex);
+            Assert.Same(item8Underlying, item8Underlying.TupleUnderlyingField);
+
+            var item9 = tuple.GetMember<TupleElementFieldSymbol>("Item9");
+            Assert.Equal("System.Int32 (System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32).Item9",
+                item9.ToTestDisplayString());
+            Assert.Equal(8, item9.TupleElementIndex);
+
+            var item9Underlying = item9.TupleUnderlyingField;
+            Assert.Equal("Item2", item9Underlying.Name);
+            Assert.IsType<SubstitutedFieldSymbol>(item9Underlying);
+            Assert.Equal(1, item9Underlying.TupleElementIndex);
+            Assert.Same(item9Underlying, item9Underlying.TupleUnderlyingField);
+        }
+
+        [Fact]
+        public void RealFieldsAreNotWrapped()
```

Comment:
>RealFieldsAreNotWrapped [](start = 20, length = 23)

:memo: I'd recommend reviewing the removal of TupleFieldSymbol by looking at this test (and following) first, as it clarifies when different types of symbols are used.